src/main/java/wis/util/InputParser.java

```diff
@@ -0,0 +1,99 @@
+package wis.util;
+
+import java.util.InputMismatchException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+
+import wis.Action;
+
+public class InputParser extends Parser {
+    public static Action parseAction(String input, String[] words) {
+        if (input.isEmpty()) {
+            return Action.NONE;
+        }
+        if (input.equals("list")) {
+            return Action.LIST;
+        }
+        if (input.equals("bye")) {
+            return Action.BYE;
+        }
+        if (words[0].equals("mark")) {
+            return Action.MARK;
+        }
+        if (words[0].equals("unmark")) {
+            return Action.UNMARK;
+        }
+        if (words[0].equals("delete")) {
+            return Action.DELETE;
+        }
+        if (words[0].equals("todo")) {
+            return Action.ADD_TODO;
+        }
+        if (words[0].equals("deadline")) {
+            return Action.ADD_DEADLINE;
+        }
+        if (words[0].equals("event")) {
+            return Action.ADD_EVENT;
+        }
+        return Action.INVALID;
```

Comment:
Perhaps use a `switch...case` block to improve readability?

```java
switch (words[0]) {
case "list":
    return Action.LIST;

case "bye":
    return Action.BYE;
...
default:
    return Action.INVALID;
```

Reply:
Thank you for suggestion. Will change this.